### PR TITLE
Extra details about i18next + better lingui examples

### PR DESCRIPTION
**Pull request by igorsantos07**
# Description

As a user of i18next for around 3 years, I noticed some parts of the comparison were lacking. I got specially baffled by the verbosity of interpolations, but I'm glad I checked the API docs before giving up :)

> BTW, do [`i18n.t()`](https://lingui.dev/ref/core#i18n.t) really need different examples from `i18n._()`? I would say that only causes more confusion; at first glance it looks like that it works differently, even though it's an alias.

An important fact to distinguish as well is that you need a lot of extra config in i18n to use natLang keys, which honestly is quite troublesome for people coming from gettext. This happens because i18next relies solely on JSON files, so they need some... _workarounds_... to shove different details in the same JSON key.

I also replaced the `t()` references with `_()`, as that's the main function (and not the alias).

Lastly, I'm not entirely sure about the syntax to include translation context in Lingui. I stumbled upon `comment` in some other example and in the API docs, but again, that's not exactly clear. I would advise add a description for all options that can be used in `_()`, as well as examples for context/comments.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update
- [x] Examples update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/lingui/js-lingui/blob/main/CONTRIBUTING.md) and [CODE_OF_CONDUCT](https://github.com/lingui/js-lingui/blob/main/CODE_OF_CONDUCT.md) docs
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation (if appropriate)

### CODE REVIEW

**Comment by igorsantos07** [comment]
By the way, [this](https://lingui.dev/tutorials/react-patterns#lazy-translations) is something that i18next does NOT support, and caused a lot of subtle bugs on our application until I realized what was going on. I'm not sure how to better write about this in the comparison page, though :|

File: website/docs/misc/i18next.md [hunk 1]
```diff
@@ -192,7 +200,19 @@ i18next.t("key", { count: 5 }); // -> "items"
...
+
+```js
+i18next.t("Hello!", { context: "This goes in the header" });
+```
+
+Lingui sample:
+
+```js
+i18n._("Hello!", { comment: "This goes in the header" });
+```
```
Comment (thekip) [inline website/docs/misc/i18next.md:215]: Unfortunately that will not work. Most lingui features build around macros, pieces of code which executed during build step not a runtime. This helps reduce size footprint of the translation catalogs as well as library itself.

In this case to specify a context for a message you MUST use a macro like this

```js
import {t} from "@lingui/macro";

t({message: `From: `, context: "direction"})
```

With plain runtime modules (i18n._) the only way is custom id.

**Comment by andrii-bodnar** [comment]
@igorsantos07 @timofei-iatsenko is there anything we can do to help move this PR forward?

**Comment by timofei-iatsenko** [comment]
This article should be proofread and lingui examples replaced to theirs `macro` alternatives.

**Comment by andrii-bodnar** [comment]
Close in favor of #2064

**Pull request closed without merging**